### PR TITLE
The deep speech demo for OpenVINO 

### PR DESCRIPTION
**Pull request by FengYen-Chang**
Hi, following the policy, I change the branch to develop, and enable the deep speech model in model downloader. Please check this, thanks.

### CODE REVIEW

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# deep-speech-v0.5.0
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Please, provide description for input and output in format like [here](https://github.com/opencv/open_model_zoo/blob/develop/models/public/mask_rcnn_resnet101_atrous_coco/mask_rcnn_resnet101_atrous_coco.md).

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@FengYen-Chang you may take a look on [PR](https://github.com/opencv/open_model_zoo/pull/433) for Open Model Zoo Contribution Guide, to check necessary steps.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md (hunk 1 above)
Comment (FengYen-Chang) [reply models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: please check the latest commit for the information of deep speech model.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+1. Audio Feature, name: `image_node`, shape: [1x16x19x26], format: [BxSxCxI],
+   where:
+
+    - B - batch size
+    - S - number of steps
+    - C - number of context
+    - I - number of input
+
+    * Note: Needs to acquire the mfcc feature for audio.
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]:
```suggestion
    > Note: Needs to acquire the mfcc feature for audio.
```
Also good idea give more explanation, for example add link to demo.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+The output contain predicted sentence for audio(speech) which decode from ctc beam search decoder. The model was trained on the LibriSpeech clean test corpus for 16-bit, 16 kHz, mono-channel WAVE audio files.
+
+
+### Converted Model
+
+The output contain predicted sentence for audio(speech) which decode from ctc beam search decoder. The model was trained on the LibriSpeech clean test corpus for 16-bit, 16 kHz, mono-channel WAVE audio files.
+
+## Legal Information
+
+[https://raw.githubusercontent.com/tensorflow/models/master/LICENSE]()
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: I think real lisense is https://raw.githubusercontent.com/mozilla/DeepSpeech/v0.5.0/LICENSE.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    - C - number of context
+    - I - number of input
+
+    * Note: Needs to acquire the mfcc feature for audio.
+
+## Output
+
+### Original Model
+
+The output contain predicted sentence for audio(speech) which decode from ctc beam search decoder. The model was trained on the LibriSpeech clean test corpus for 16-bit, 16 kHz, mono-channel WAVE audio files.
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Please provide more information, such as name and shape.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+| MParams                         | 47.225                                    |
+| Source framework                | TensorFlow\*                              |
+
+## Performance
+
+## Input
+
+### Original Model
+
+1. Audio, name: `input_samples`, a WAVE file.
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: There are three input layer in `model.yml` : `input_node`, `previous_state_h/read`, `previous_state_c/read`. Please specify all of them.

(also three outputs)

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md (hunk 3 above)
Comment (FengYen-Chang) [reply models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Thanks a lot :)

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md (hunk 5 above)
Comment (FengYen-Chang) [reply models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Thanks for your review. I have updated the file, please check it again, thanks.

File: models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+    - B - batch size
+    - F - number of features in the hidden state `c`
+    
+    > Note: The initial value of hidden state `c` is zero.
+
+## Output
+
+### Original Model
+
+Output text, name: `Softmax`, Contains predicted sentence for audio(speech) which decode from ctc beam search decoder. The model was trained on the LibriSpeech clean test corpus for 16-bit, 16 kHz, mono-channel WAVE audio files.
```
Comment (eizamaliev) [inline models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Please, make the output layers numbered list, like in `Input` section.

Comment (FengYen-Chang) [reply models/public/deep-speech-v0.5.0/deep-speech-v0.5.0.md]: Thanks for your review :)

**Comment by FengYen-Chang** [comment]
#419

**Comment by FengYen-Chang** [comment]
Hi @vladimir-dudnik, I have implemented the accuracy checker for deep speech, could you help me to check this?

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
@FengYen-Chang when I try to run demo with 1272-128104-000.wav file error message says that there is no Softmax key. If I change 'Softmax' in deepspeech_openvino_0.5.py, line 178 to 'raw_logits' (what seems to be an output blob) then demo run with runtime warnings but recognize the phrase. Could you please fix runtime warnings?

[ INFO ] Creating Inference Engine
[ INFO ] Loading network files:
	/home/zvdudnik/dev/data/OMZ/PR/models/public/deep-speech-v0.5.0/FP32/deep-speech-v0.5.0.xml
	/home/zvdudnik/dev/data/OMZ/PR/models/public/deep-speech-v0.5.0/FP32/deep-speech-v0.5.0.bin
[ INFO ] Preparing input blobs
[ INFO ] Preparing output blobs
[ INFO ] Loading model to the plugin
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
[ INFO ] Processing output blob
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:45: RuntimeWarning: overflow encountered in double_scalars
  _TpB = _pT['l'][_candidate] * pred[_t][idx_b]
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:60: RuntimeWarning: overflow encountered in double_scalars
  _TpNB = v * _pT['l'][_candidate]
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:57: RuntimeWarning: overflow encountered in double_scalars
  _TpNB = v * _pB['l'][_candidate]
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:44: RuntimeWarning: overflow encountered in double_scalars
  _TpNB = _pNB['l'][_candidate] * pred[_t][_candidate[-1]]
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:63: RuntimeWarning: invalid value encountered in double_scalars
  _pT['c'][extand_t] += _TpNB
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:64: RuntimeWarning: invalid value encountered in double_scalars
  _pNB['c'][extand_t] += _TpNB
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:47: RuntimeWarning: invalid value encountered in double_scalars
  _pNB['c'][_candidate] += _TpNB
/home/zvdudnik/dev/data/OMZ/PR/open_model_zoo/demos/python_demos/deep_speech_demo/ctc_beamsearch_decoder.py:51: RuntimeWarning: invalid value encountered in double_scalars
  _pT['c'][_candidate] = _pNB['c'][_candidate] + _pB['c'][_candidate]

>>>mister cuilter is the ampostle of the middle classes and we are glad to welcome his ghospelh

**Comment by FengYen-Chang** [comment]
Hi @vladimir-dudnik, Thanks for your testing, I have changed the node from `Softmax` to `raw_logits`, this error cause by different MO version. Since the MO version is different, the output are different and cause the warnings happed. To resolve this, I have added the softmax function.

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by FengYen-Chang** [comment]
Hi @vladimir-dudnik, I found I made an error when converting the model automatically, so I patch the `model.yml` to change the output node from `raw_logits` to `Softmax`.

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a built

**Comment by vladimir-dudnik** [comment]
Jenkins please retry a build

File: demos/python_demos/deep_speech_demo/README.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+```sh
+python3 ./mo_tf.py
+--input_model path_to_model/output_graph.pb                         \
+--freeze_placeholder_with_value input_lengths->[16]                 \
+--input input_node,previous_state_h/read,previous_state_c/read      \
+--input_shape [1,16,19,26],[1,2048],[1,2048]                        \
+--output raw_logits,lstm_fused_cell/GatherNd,lstm_fused_cell/GatherNd_1 \ 
+--disable_nhwc_to_nchw
+```
+
```
Comment (eizamaliev) [inline demos/python_demos/deep_speech_demo/README.md]: ... or get it from Open Model Zoo using `downloader.py` and `converter.py`

Comment (IRDonch) [reply demos/python_demos/deep_speech_demo/README.md]: Not even "or", I would say. If the model is downloadable/convertible with our scripts, the README should just tell the user to use them. No need to duplicate this information.

File: models/public/deep-speech-v0.5.0/model.yml [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+description: >-
+  The `deep-speech` model is a topology intended to perform speech to text.
+  This model is implemented using the tensorflow\* framework.
```
Comment (IRDonch) [inline models/public/deep-speech-v0.5.0/model.yml]: TensorFlow

File: demos/python_demos/deep_speech_demo/speech_features.py [hunk 9]
```diff
@@ -0,0 +1,199 @@
+##################################################
+#
+# John Feng, 2019/09/06 
+#
+##################################################
```
Comment (IRDonch) [inline demos/python_demos/deep_speech_demo/speech_features.py]: Please remove these banners and add the standard OMZ copyright banners instead. (You already have the correct banner in the main script.)

File: demos/python_demos/deep_speech_demo/deep_speech_demo.py [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+beamwidth  = 10
+
+def main():
+    log.basicConfig(format="[ %(levelname)s ] %(message)s", level=log.INFO, stream=sys.stdout)
+    args = build_argparser().parse_args()
+    
+    alphabet = Alphabet(os.path.abspath(args.alphabet))
+    
+    # Speech feature extration
+    _wave = wave.open('1272-128104-0000.wav', 'rb')
```
Comment (IRDonch) [inline demos/python_demos/deep_speech_demo/deep_speech_demo.py]: I think you forgot to use the `-i` option here. Also, please use `with` to ensure the file is closed.

File: demos/python_demos/deep_speech_demo/deep_speech_demo.py [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+    fs = _wave.getframerate()
+    if fs is not 16000:
+        log.error("Please using 16kHz wave file, not {}\n".format(fs))
+    _length = _wave.getnframes()
+    audio = np.zeros(_length)
+
+    for i in range(_length):
+        audio[i] = int(struct.unpack("<h",_wave.readframes(1))[0])
```
Comment (IRDonch) [inline demos/python_demos/deep_speech_demo/deep_speech_demo.py]: Reading frames one at a time is inefficient. You could replace this with `audio = np.frombuffer(_wave.readframes(_length), dtype=np.dtype('<h'))`.

**Comment by FengYen-Chang** [comment]
Close the PR for some issues.
